src/python/aqueduct_executor/operators/airflow/dag.template

```diff
@@ -1,100 +1,80 @@
 from datetime import datetime
-from airflow.models import DAG
-
-from airflow.operators.python import PythonOperator
 
-from aqueduct_executor.operators.function_executor import execute as func_execute
-from aqueduct_executor.operators.function_executor import spec as func_spec
-from aqueduct_executor.operators.param_executor import execute as param_execute
-from aqueduct_executor.operators.param_executor import spec as param_spec
-from aqueduct_executor.operators.system_metric_executor import execute as sys_metric_execute
-from aqueduct_executor.operators.system_metric_executor import spec as sys_metric_spec
-from aqueduct_executor.operators.connectors.data import execute as conn_execute
-from aqueduct_executor.operators.connectors.data import spec as conn_spec
-from aqueduct_executor.operators.utils import enums
+from airflow.models import DAG
+from airflow.operators.python import PythonVirtualenvOperator
+
+# Python requirements for each Airflow task
+VENV_REQUIREMENTS=[
+    "aqueduct-ml",
+    "awswrangler",
+    "boto3",
+    "cloudpickle",
+    "distro",
+    "google-cloud-bigquery",
+    "mysqlclient",
+    "pandas",
+    "psycopg2-binary",
+    "pyarrow",
+    "pydantic",
+    "pyodbc",
+    "pyyaml",
+    "snowflake-sqlalchemy",
+    "SQLAlchemy",
+    "typing_extensions",
+]
 
 def invoke_task(spec, **kwargs):
     '''
     Check the spec type and invoke the correct operator.
     First, append the dag_run_id to all of the storage paths in the spec.
     '''
+    from aqueduct_executor.operators.utils import enums
+    from aqueduct_executor.operators.function_executor import spec as func_spec
+    from aqueduct_executor.operators.param_executor import spec as param_spec
+    from aqueduct_executor.operators.system_metric_executor import spec as sys_metric_spec
+    from aqueduct_executor.operators.connectors.data import spec as conn_spec
+
+    from aqueduct_executor.operators.function_executor import execute as func_execute
+    from aqueduct_executor.operators.param_executor import execute as param_execute
+    from aqueduct_executor.operators.system_metric_executor import execute as sys_metric_execute
+    from aqueduct_executor.operators.connectors.data import execute as conn_execute
+
     dag_run_id = kwargs["run_id"]
 
     spec_type = enums.JobType(spec["type"])
     if spec_type == enums.JobType.FUNCTION:
         spec = func_spec.FunctionSpec(**spec)
-        handle_function(spec, dag_run_id)
+        spec.metadata_path = "{}_{}".format(spec.metadata_path, dag_run_id)
```

Comment:
While this is less human friendly to read, it was necessary to remove the helper functions, because of how Python code is executed inside of the Airflow PythonVirtualenv operator.

Any imports or helper functions must be defined inside of the Python `callable` that is specified in the Airflow task definition below. This means that any other imports or helper functions will not be initialized by the Python runtime. This was not true in the non virtualenv operators, because the entire file was first initialized by the Python runtime.